TELCODOCS#2004: Day 2 Operations - Security Doc (#86603)
openshift/openshift-docs: modules/telco-security-rhcos-overview.adoc

Reviewer: Right below the "Console access" section we should make a NOTE that says:
Please follow your company standards for securing console access

Author: Updated as per the suggestion.

// Module included in the following assemblies:
//

> // * edge_computing/day_2_core_cnf_clusters/observability/telco-security-host-sec.adoc

Reviewer 2: ```suggestion
// * edge_computing/day_2_core_cnf_clusters/security/telco-security-host-sec.adoc

Reviewer 2: I didn't check all of these ^, just noticed this one.

Author: Updated as per the suggestion. Also updated other modules.

:_mod-docs-content-type: CONCEPT
[id="telco-security-rhcos-overview_{context}"]

> = Red Hat Enterprise Linux CoreOS (RHCOS)

Author: Updated as per the suggestion.

> Red Hat Enterprise Linux CoreOS (RHCOS) is different from Red Hat Enterprise Linux (RHEL) in key areas. For more information, see "About RHCOS".

Author: Updated as per the suggestion.

From a telco perspective, a major distinction is the control of `rpm-ostree`, which is updated through the Machine Config Operator. 

> {op-system} follows the same immutable design used for pods in {product-title}. This ensures that the operating system remains consistent across the cluster. For information about {op-system} architecture, see "{op-system-first}".

Reviewer 2: I might be being too picky here, but it seems odd to have me see a section that has the same name as the one I am in. How about:

```suggestion
{op-system} follows the same immutable design used for pods in {product-title}. This ensures that the operating system remains consistent across the cluster. For information about {op-system} architecture, see "{op-system-first} in _Architecture_.
```

Author: Based on the guidelines to add xrefs in modules, I think just mentioning the section name wrapped in "" is the right way to do it. As we have provided an xref, this should not cause any confusion. 
To maintain consistency with similar instances, keeping it as is.

> To manage hosts effectively while maintaining security, avoid direct access whenever possible. It is best to avoid direct access to hosts. Instead, you can use methods for host management:

Author: Updated as per the suggestion.

* Debug pod
* Direct SSHs
* Console access

> RHCOS was built with a strong emphasis on security. The following security mechanisms are integral to maintaining host security:

Author: Updated as per the suggestion.

> Linux namespaces:: Provide isolation for processes and resources. Each container's processes and files are kept within its own namespace. If a user escapes from the container’s namespace, they could gain access to the host operating system, which can compromise security.

Author: Updated as per the suggestion.

Security-Enhanced Linux (SELinux):: Enforces mandatory access controls to restrict access to files and directories by processes. It adds an extra layer of security by preventing unauthorized access to files if a process tries to break its confinement.
+
SELinux follows the security policy of denying everything unless explicitly allowed. If a process attempts to modify or access a file without permission, SELinux denies access. For more information, see link:https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html-single/using_selinux/index#introduction-to-selinux_getting-started-with-selinux[Introduction to SELinux].

> Linux capabilities:: Assigns specific privileges to processes at a granular level, minimizing the need for full root permissions. For more information, see "Linux capabilities".

Reviewer 2: ```suggestion
Linux capabilities:: Assign specific privileges to processes at a granular level, minimizing the need for full root permissions. For more information, see "Linux capabilities".
```

Author: Updated as per the suggestion.

> Control group (cgroup):: Allocates and manages system resources, such as CPU and memory for processes and containers, ensuring efficient usage. As of {product-title} 4.16, there are two versions of cgroups. Cgroup v2 is now configured by default.

Reviewer 2: “cgroup” stands for “control group” and is never capitalized. 
```suggestion
Control groups (cgroups):: Allocate and manage system resources, such as CPU and memory for processes and containers, ensuring efficient usage. As of {product-title} 4.16, there are two cgroups. cgroup v2 is now configured by default.
```

Author: Updated as per the suggestion.

> Control group (cgroup):: Allocates and manages system resources (CPU, memory, and so on) for processes and containers, ensuring efficient usage. it ensures fair allocation of resources across processes and containers. As of {product-title} 4.16, there are two versions of cgroups. Cgroup v2 is now configured by default.

Author: Updated as per the suggestion.
